web/ui/react-app/src/QueryStats.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { PureComponent } from 'react';
+import './QueryStats.css';
+
+interface QueryStatsProps {
+    stats: {
+        loadTime: number;
+        resolution: number;
+        totalTimeSeries: number;
+    } | null;
+    loading: boolean;
+}
+
+class QueryStats extends PureComponent<QueryStatsProps>{
```

Comment:
I would suggest you convert it to Functional component since is not maintaining any state or dealing with any of the lifecycles hooks. If you want to preserve it like PureComponent just use memo from React